tests/90jira/SYN-205.pl

```diff
@@ -11,14 +11,25 @@
       ->then( sub {
          my ( $room_id ) = @_;
 
-         await_event_for( $invitee, filter => sub {
-            my ( $event ) = @_;
-            return unless $event->{type} eq "m.room.member" and
-                          $event->{room_id} eq $room_id and
-                          $event->{state_key} eq $invitee->user_id and
-                          $event->{content}{membership} eq "invite";
+         await_sync($invitee, check => sub {
+            my ( $sync_body ) = @_;
+            log_if_fail $sync_body;
+            my $room =  $sync_body->{rooms}{invite}{$room_id};
+            return unless $room;
+            assert_json_keys( $room, qw( invite_state ) );
+            assert_json_keys( $room->{invite_state}, qw( events ) );
+            my $invite = first {
+               $_->{type} eq "m.room.member"
+                  and $_->{state_key} eq $invitee->user_id
+            } @{ $room->{invite_state}{events} };
 
-            return 1;
-         })->SyTest::pass_on_done( "Invitee received invite event" )
+            assert_json_keys( $invite, qw( sender content state_key type ));
+            $invite->{content}{membership} eq "invite"
+               or die "Expected an invite event";
+            $invite->{sender} eq $user->user_id
+               or die "Expected the invite to be from user A";
```

Comment:
why not use `assert_eq` here?

Reply:
Mainly because the code before had it in this format. I don't really care which we end up using.

Reply:
which code?

Reply:
I yanked `invite_state` code from invite tests when mapping this over to use `/sync`.

Reply:
I'm still unclear on where you've cargo-culted this from, but it doesn't really matter. The situation is:

 * yes, there exists code which uses the `$foo eq $bar or die` pattern
 * that code either pre-dates the existence of `assert_eq`, or has been cargo-culted from other code that does
 * I'd prefer it if we didn't proliferate the cargo-culting, but I'm not going to insist on it.